nntrainer/src/layer.cpp

```diff
@@ -171,4 +178,42 @@ int Layer::setProperty(std::vector<std::string> values) {
   return status;
 }
 
+int Layer::setName(std::string name) {
+  int status = ML_ERROR_NONE;
+  std::set<std::string>::iterator iter;
+
+  if (name == this->name)
+    return status;
+
+  iter = layer_names.find(name);
+  if (name.empty()) {
+    status = ML_ERROR_INVALID_PARAMETER;
```

Comment:
is this right? name.empty()? if it is right, then it should be checked earlier than this. I guess.

Reply:
Updated.